xds/internal/server/listener_wrapper.go

```diff
@@ -331,15 +331,19 @@ func (l *listenerWrapper) handleRDSUpdate(update rdsHandlerUpdate) {
 	}
 	if update.err != nil {
 		if xdsresource.ErrType(update.err) == xdsresource.ErrorTypeResourceNotFound {
-			l.switchMode(nil, connectivity.ServingModeNotServing, update.err)
+			l.mu.Lock()
+			l.filterChains = nil
```

Comment:
This is wrong; we should not throw away the LDS data when we get a bad RDS.  A good RDS needs to be able to work again.